go.mod

```diff
@@ -1,6 +1,6 @@
 module sigs.k8s.io/cluster-api
 
-go 1.20
+go 1.21
```

Comment:
WDYT about also setting toolchain like CR? k/k didn't do it.

After reading https://go.dev/doc/toolchain I'm not sure if we should or not

(https://github.com/kubernetes-sigs/controller-runtime/blob/52c391f17373684a96fa3b552d8a77b9da8ae5b5/go.mod#L5)

Reply:
> The go and toolchain lines can be thought of as specifying the version requirements for the module's dependency on the Go toolchain itself, just as the require lines in go. mod specify the version requirements for dependencies on other modules.

Based on this it sounds reasonable to set it

Reply:
Although we should maybe just set 1.21.5

Reply:
Setting toolchain to 1.21 would **force** folks who import this or newer versions of capi to use go 1.21.

Reply:
I think we already do it via `go 1.21`.

Reply:
Talked to Christian about it. We don't see a strong reason why we should set `toolchain`. We already enforce usage of Go 1.21 via `go 1.21` and we probably don't want to enforce a specific minimum go version if we don't have a strong reason to do so.

Thus I would merge as is. If someone comes up with a reason why we should set it, let us know and we follow-up